pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -30,17 +30,143 @@ import (
 	"istio.io/istio/pilot/pkg/networking/plugin"
 	"istio.io/istio/pilot/pkg/networking/util"
 	"istio.io/istio/pilot/pkg/security/model"
+	"istio.io/istio/pkg/config/constants"
 	"istio.io/istio/pkg/config/mesh"
 	"istio.io/istio/pkg/proto"
 )
 
 func TestBuildGatewayListenerTlsContext(t *testing.T) {
 	testCases := []struct {
-		name      string
-		server    *networking.Server
-		enableSds bool
-		result    *auth.DownstreamTlsContext
+		name             string
+		server           *networking.Server
+		enableIngressSds bool
+		sdsConfig        model.WorkloadSdsConfig
+		result           *auth.DownstreamTlsContext
 	}{
+		{
+			name: "no ingress sds, mesh SDS disabled, tls mode ISTIO_MUTUAL",
+			server: &networking.Server{
+				Hosts: []string{"httpbin.example.com"},
+				Tls: &networking.Server_TLSOptions{
+					Mode: networking.Server_TLSOptions_ISTIO_MUTUAL,
+				},
+			},
+			enableIngressSds: false,
+			result: &auth.DownstreamTlsContext{
+				CommonTlsContext: &auth.CommonTlsContext{
+					AlpnProtocols: util.ALPNHttp,
+					TlsCertificates: []*auth.TlsCertificate{
+						{
+							CertificateChain: &core.DataSource{
+								Specifier: &core.DataSource_Filename{
+									Filename: constants.DefaultCertChain,
+								},
+							},
+							PrivateKey: &core.DataSource{
+								Specifier: &core.DataSource_Filename{
+									Filename: constants.DefaultKey,
+								},
+							},
+						},
+					},
+					ValidationContextType: &auth.CommonTlsContext_ValidationContext{
+						ValidationContext: &auth.CertificateValidationContext{
+							TrustedCa: &core.DataSource{
+								Specifier: &core.DataSource_Filename{
+									Filename: constants.DefaultRootCert,
+								},
+							},
+						},
+					},
+				},
+				RequireClientCertificate: proto.BoolTrue,
+			},
+		},
+		{
+			name: "no ingress sds, mesh SDS enabled, tls mode ISTIO_MUTUAL",
```

Comment:
how is this going to work? if there is no sds agent in the ingress gateway

Reply:
This test was written with `egress-gateway` in mind, but even without ingress sds agent, the gateway would be able to access the nodeagent sds service if sds is globally enabled (interestingly, sds enabled `ingress-gateways` uses a separate pod-embedded node agent).